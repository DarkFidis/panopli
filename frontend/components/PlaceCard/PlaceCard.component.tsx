import React from "react";
import styles from "../../styles/Card.module.css";
import {CardProps} from "../../types/CardProps";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faStar } from "@fortawesome/free-solid-svg-icons";

export const PlaceCard: React.FC<CardProps> = ({ imgUrl, title, stars, restoUrl}) => {
  const arrayOfStars = Array.from(Array(stars).keys())
  return (
    <div className={styles.card}>
      <img src={imgUrl} alt=""/>
      <div className={styles.content}>
        <h3 className={styles.title}>{title}</h3>
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam animi consequuntur ea molestias illo.
          Porro error facilis obcaecati. Quisquam, placeat.</p>
        <div className={styles.stars}>
          { arrayOfStars.map((_, key) => (
            <FontAwesomeIcon icon={faStar} key={key} className={styles.stars} />
          ))}
        </div>
        <a href={restoUrl} target='_blank'>
          <button className={styles.btn}>check out!</button>
        </a>
      </div>
    </div>
  )
}

export default PlaceCard
