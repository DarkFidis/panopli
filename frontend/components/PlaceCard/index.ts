import PlaceCardComponent, { PlaceCard } from "./PlaceCard.component";
export { PlaceCard }
export default PlaceCardComponent
