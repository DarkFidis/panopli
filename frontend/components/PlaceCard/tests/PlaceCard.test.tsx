import * as React from 'react'

import { render } from "@testing-library/react";
import PlaceCard from "../index";

describe('PlaceCard component unit test', () => {
  it('should match snapshot when inactive', () => {
    // Given
    const props = {
      active: false,
      place: {"location":{"type":"Point","coordinates":[48.86172,2.342443]},"properties":{"address":{"street":"121 Rue Saint-Honoré","postalCode":"75001","city":"Paris"},"website":"https://www.yamtcha.com/","type":"b","name":"Yam'Tcha","rating":4.6},"type":"Feature"},
      setActivePlace: () => {}
    }
    const { container } = render(<PlaceCard {...props} />)
    expect(container).toMatchSnapshot()
  })
  it('should match snapshot when active', () => {
    // Given
    const props = {
      active: true,
      place: {"location":{"type":"Point","coordinates":[48.86172,2.342443]},"properties":{"address":{"street":"121 Rue Saint-Honoré","postalCode":"75001","city":"Paris"},"website":"https://www.yamtcha.com/","type":"b","name":"Yam'Tcha","rating":4.6},"type":"Feature"},
      setActivePlace: () => {}
    }
    const { container } = render(<PlaceCard {...props} />)
    expect(container).toMatchSnapshot()
  })
})
