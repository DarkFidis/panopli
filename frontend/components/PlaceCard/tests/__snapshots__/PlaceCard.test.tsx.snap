// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard component unit test should match snapshot when active 1`] = `
<div>
  <div
    class="card "
  >
    <img
      alt=""
      src="images/resto-3.jpeg"
    />
    <div
      class="content"
    >
      <h3
        class="title"
      >
        Yam'Tcha
      </h3>
      <p>
        121 Rue Saint-Honoré
      </p>
      <p>
        75001
          
        Paris
      </p>
      <div
        class="stars"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <p>
          4.6
        </p>
      </div>
      <a
        href="https://www.yamtcha.com/"
      >
        <button
          class="btn"
        >
          Site web
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`PlaceCard component unit test should match snapshot when inactive 1`] = `
<div>
  <div
    class="card "
  >
    <img
      alt=""
      src="images/resto-3.jpeg"
    />
    <div
      class="content"
    >
      <h3
        class="title"
      >
        Yam'Tcha
      </h3>
      <p>
        121 Rue Saint-Honoré
      </p>
      <p>
        75001
          
        Paris
      </p>
      <div
        class="stars"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star stars"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
          />
        </svg>
        <p>
          4.6
        </p>
      </div>
      <a
        href="https://www.yamtcha.com/"
      >
        <button
          class="btn"
        >
          Site web
        </button>
      </a>
    </div>
  </div>
</div>
`;
